### ISSUE TITLE
Docker service failed to start after upgraded to 1548.2.0 

### ISSUE THREAD

**Issue by xueshanf** [coreos/bugs#2195]
# Issue Report #

docker service fail to start

### Container Linux Version ###

```
$ cat /etc/os-release
NAME="Container Linux by CoreOS"
ID=coreos
VERSION=1548.2.0
VERSION_ID=1548.2.0
BUILD_ID=2017-10-12-0514
PRETTY_NAME="Container Linux by CoreOS 1548.2.0 (Ladybug)"
ANSI_COLOR="38;5;75"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://issues.coreos.com"
COREOS_BOARD="amd64-usr"
...
BUG_REPORT_URL="https://issues.coreos.com"
```

### Environment ###

AWS 
### Expected Behavior ###

Docker service should start after upgraded to  1548.2.0 (beta).

### Actual Behavior ###

Docker service should start.

### Other Information ###

  1 After reboot:

```
● docker.socket - Docker Socket for the API
   Loaded: loaded (/run/systemd/system/docker.socket; enabled; vendor preset: disabled)
   Active: failed (Result: service-start-limit-hit) since Fri 2017-10-13 18:45:29 UTC; 3s ago
   Listen: /var/run/docker.sock (Stream)

Oct 13 18:45:29 ip-10-0-5-171.us-west-2.compute.internal systemd[1]: Starting Docker Socket for the API.
Oct 13 18:45:29 ip-10-0-5-171.us-west-2.compute.internal systemd[1]: Listening on Docker Socket for the API.
Oct 13 18:45:29 ip-10-0-5-171.us-west-2.compute.internal systemd[1]: docker.socket: Unit entered failed state.
```
  2. docker info

```
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
```

  3. systemctl status docker.service

```
systemctl status docker.service
● docker.service - Docker Application Container Engine
   Loaded: loaded (/run/systemd/system/docker.service; disabled; vendor preset: disabled)
  Drop-In: /etc/systemd/system/docker.service.d
           └─50-docker-wait-for-device.conf, 60-docker-wait-for-var-lib.conf
   Active: failed (Result: exit-code) since Fri 2017-10-13 18:47:16 UTC; 34s ago
     Docs: http://docs.docker.com
  Process: 2093 ExecStart=/usr/bin/env PATH=${TORCX_BINDIR}:${PATH} ${TORCX_BINDIR}/dockerd --host=fd:// --containerd=/var/run/docker/libcontainerd/docker-containerd.sock $DOCKER_SELINUX $DOCKER_
 Main PID: 2093 (code=exited, status=1/FAILURE)
      CPU: 16ms

Oct 13 18:47:11 ip-10-0-5-171.us-west-2.compute.internal systemd[1]: docker.service: Main process exited, code=exited, status=1/FAILURE
Oct 13 18:47:11 ip-10-0-5-171.us-west-2.compute.internal systemd[1]: Failed to start Docker Application Container Engine.
Oct 13 18:47:11 ip-10-0-5-171.us-west-2.compute.internal systemd[1]: docker.service: Unit entered failed state.
Oct 13 18:47:11 ip-10-0-5-171.us-west-2.compute.internal systemd[1]: docker.service: Failed with result 'exit-code'.
Oct 13 18:47:16 ip-10-0-5-171.us-west-2.compute.internal systemd[1]: docker.service: Service hold-off time over, scheduling restart.
Oct 13 18:47:16 ip-10-0-5-171.us-west-2.compute.internal systemd[1]: Stopped Docker Application Container Engine.
Oct 13 18:47:16 ip-10-0-5-171.us-west-2.compute.internal systemd[1]: docker.service: Start request repeated too quickly.
Oct 13 18:47:16 ip-10-0-5-171.us-west-2.compute.internal systemd[1]: Failed to start Docker Application Container Engine.
Oct 13 18:47:16 ip-10-0-5-171.us-west-2.compute.internal systemd[1]: docker.service: Unit entered failed state.
Oct 13 18:47:16 ip-10-0-5-171.us-west-2.compute.internal systemd[1]: docker.service: Failed with result 'exit-code'.
```

### Workaround

Add a drop-in for docker.service under /etc/system/system/docker.service.d/50-docker-opts.conf with:

```
[Service]
Environment=DOCKER_OPTS='--storage-driver overlay'
```
Then `systemctl daemon-reload; systemctl restart docker.socket`.  This fixed the problem.

```
$ docker info
Containers: 1
 Running: 1
 Paused: 0
 Stopped: 0
Images: 4
Server Version: 1.12.6
Storage Driver: overlay
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp selinux
Kernel Version: 4.13.5-coreos-r2
Operating System: Container Linux by CoreOS 1548.2.0 (Ladybug)
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 1.955 GiB
Name: ip-10-0-5-171.us-west-2.compute.internal
ID: BXH7:GJ4N:ZBP7:DDJW:VKTP:LSFV:A24N:ATII:74SK:QEVS:PBNJ:2JAO
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Username: suharbormaster
Registry: https://index.docker.io/v1/
```

**Comment by euank** [comment]
Do you have `journalctl` output for when docker failed? Is it complaining about multiple graphdrivers being present?

I can't trivially reproduce this by upgrading from the previous beta.
I launched ami-52bd412a (v1520.4.0, the previous beta) and ran a few containers, then updated to 1548.2.0, and all still seemed well.

Do you know what version you originally started on (which can sometimes be determined via `curl 169.254.169.254/latest/meta-data/ami-id`)?

Do you have any more information about how to reproduce this?

**Comment by xueshanf** [comment]
@euankYou nailed it!  We had a configuration problem. The EC2 was built with alpha channel, but the cloud-config has the update group beta channel! 

The journalclt during reboot reports multiple graphdrivers like so:
```
Oct 12 05:30:55 ip-10-0-5-187.us-west-2.compute.internal env[1042]: time="2017-10-12T05:30:55.581776171Z" level=error msg="Failed to built-in GetDriver graph aufs /var/lib/docker"
Oct 12 05:30:55 ip-10-0-5-187.us-west-2.compute.internal env[1042]: time="2017-10-12T05:30:55.602308143Z" level=info msg="Graph migration to content-addressability took 0.00 seconds"
Oct 12 05:30:56 ip-10-0-5-187.us-west-2.compute.internal env[1042]: time="2017-10-12T05:30:56.364368601Z" level=info msg="Docker daemon" commit=30311de graphdriver=overlay version=1.12.6
Oct 13 05:30:55 ip-10-0-5-187.us-west-2.compute.internal env[1023]: time="2017-10-13T05:30:55.895359180Z" level=fatal msg="Error starting daemon: error initializing graphdriver: \"/var/lib/docker\" contains several valid graphdrivers: overlay, overlay2; Please cleanup or explicitly choose storage driver (-s <DRIVER>)"
```

The previous image before the upgrade was ami-4f47b937,  CoreOS-alpha-1548.0.0-hvm.  I guess what's what caused the problem! The EC2 started out with alpha image, but the system updated with beta channel image. Sorry about the false alarm. I will correct the error and rebuild.

**Comment by euank** [comment]
Indeed, right now going from alpha to beta/stable results in `docker` downgrading from 17.xx to 1.12.. which isn't supported by Docker and so has a chance to break like that.

I realized that we have instructions for changing upgrade channels, but haven't made it clear that going "backwards" from alpha to beta can cause problems. I've filed https://github.com/coreos/bugs/issues/2198 to update our docs to make that clear.

Thanks for filing the issue; do let me know if you run into any other problems.
